packages/devtools_app/lib/src/ui/search.dart

```diff
@@ -605,8 +605,6 @@ mixin SearchFieldMixin<T extends StatefulWidget> on State<T> {
   /// pop-up is visible close the pop-up on first ESCAPE.
   /// [keyEventsToPropogate] a set of key events that should be propogated to
   /// other handlers
-  /// TODO(elliette): Have this return a private _AutoCompleteSearchField widget
-  ///  instead.
   Widget buildAutoCompleteSearchField({
```

Comment:
Is there a reason all of this logic can't be in `_AutoCompleteSearchField`? Then where ever `buildAutoCompleteSearchField` is being called you would instead use the `_AutoCompleteSearchField` widget directly.

Reply:
Or actually, now I'm remembering. We only want classes that mixin SearchFieldMixin to be able to access these widgets. Same comment applies, except `buildAutoCompleteSearchField` would instead return `_AutoCompleteSearchField` directly and all the business logic can be part of `_AutoCompleteSearchField`